directembedding/src/main/scala/ch/epfl/directembedding/transformers/EmbeddingTransformation.scala

```diff
@@ -98,47 +130,44 @@ trait ReifyAsEmbedding extends DirectEmbeddingModule
     private def selfType(tree: Tree): Type = {
       val owner = tree.symbol.owner
       if (owner.isType) owner.asType.toType
-      else throw new RuntimeException(s"Unable to find type of $tree")
+      else c.abort(tree.pos, s"Unable to resolve type for $tree")
     }
 
-    private def fallbackAnnotation(treeSymbol: Symbol, tree: Tree): Option[Annotation] =
-      for {
+    private def fallbackAnnotation(treeSymbol: Symbol, tree: Tree): Iterable[Annotation] = {
+      val sym = for {
         fallbackType <- typeMap.get(selfType(tree).typeSymbol.fullName)
         fallbackSymbol <- {
           log(s"Falling back to $fallbackType")
           findMatchingSymbol(treeSymbol, fallbackType)
         }
-        fallbackAnnotation <- {
-          log(s"Found matching symbol in overridden type: $fallbackSymbol")
-          reifyAnnotation(fallbackSymbol)
-        }
-      } yield fallbackAnnotation
+      } yield fallbackSymbol
+      sym.toList.flatMap(_.annotations)
+    }
 
     /**
-     * Get body of class to be reified through [[reifyAs]] annotation.
+     * Strip away sugar from partially applied functions in annotations
      *
-     * First, performs lookup for the [[reifyAs]] annotation in the invoked
-     * symbol and falls back to a lookup in [[typeMap]] for the type of
-     * parameter `tree` if no annotation exists.
+     * Example:
+     *   @reifyAs(method _)
      *
-     * @param lhs Tree that was invoked
-     * @param tree Tree to be reified through annotation
-     * @return Reified tree if annotation exists, fails compilation otherwise
+     * is sugar for:
+     *   {(args: Args*) => method(args)}
      */
-    private def body(lhs: Tree, tree: Tree): Tree = {
-      val treeSymbol = symbol(lhs)
-      reifyAnnotation(treeSymbol).orElse {
-        log(s"Missing reifyAs annotation for $treeSymbol")
-        fallbackAnnotation(treeSymbol, tree)
-      }.map {
-        _.tree.children.tail.headOption.getOrElse {
-          throw new IllegalArgumentException("Missing argument for reifyAs annotation")
-        }
-      }.getOrElse {
-        c.abort(tree.pos, s"$treeSymbol on ${selfType(tree).typeSymbol} is not supported in $dslName")
+    private def stripAnonFunctions(tree: Tree): Option[Tree] = {
+      tree match {
```

Comment:
Less code is good and easy to understand. We need to keep the code as clean and compact as possible.